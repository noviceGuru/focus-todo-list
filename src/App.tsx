import { useState } from "react"

import MainTodo from "components/atoms/main-todo/main-todo"
import Overlay from "components/atoms/overlay/overlay"
import BackTable from "components/molecules/back-table/back-table"

import OpenEnvelope from "assets/open-envelope.svg"
import ClosedEnveloper from "assets/closed-envelope.svg"

export type Todo = {
    id: string
    task: string
}

const initialTodos : Todo[] = [
    {
        id: "1",
        task: "task one",
    },
    {
        id: "2",
        task: "task two",
    },
    {
        id: "3",
        task: "task three",
    },
]


function App() {
    const [show, setShow] = useState<boolean>(false)

    const [todos, setTodos] = useState<Todo[]>(initialTodos)

    return (
        <div className="flex flex-col justify-center items-center h-full">
            <MainTodo {...todos[0]} setTodos={setTodos} />
            <Overlay show={show} setShow={setShow}>
                <BackTable todos={todos} setTodos={setTodos}/>
            </Overlay>
            <img
                src={show ? OpenEnvelope : ClosedEnveloper}
                onClick={() => setShow(true)}
                className="absolute h-12 bottom-14 md:bottom-36 cursor-pointer"
            />
        </div>
    )
}

export default App
