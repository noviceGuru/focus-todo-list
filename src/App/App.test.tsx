import { test, expect, afterEach } from "vitest"
import App from "./App"
import { render, screen, fireEvent, cleanup } from "@testing-library/react"

afterEach(()=>cleanup())

test("The empty screen shows the empty note at the beginning.", () => {
    render(<App />)
    const allDone = screen.getByText(/all done!/i)

    expect(allDone).toBeTruthy()
})

test("onClick on the envelope, the overlay opens, matching the snapshot.", () => {
    render(<App />)
    const envelopIcon = screen.getByRole("img", {
        name: /envelop\-icon/i,
    })

    fireEvent.click(envelopIcon)

    expect(screen.getByTestId('pop-up-dialog')).toBeTruthy()
    expect(screen.getByTestId('pop-up-dialog')).toMatchSnapshot()
})