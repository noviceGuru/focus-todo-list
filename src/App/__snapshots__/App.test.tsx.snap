// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`onClick on the envelope, the overlay opens, matching the snapshot. 1`] = `
<div
  class="fixed z-20 inset-0 bg-gray-400 bg-opacity-50 flex items-center justify-center cursor-pointer [&>*]:cursor-default"
  data-testid="pop-up-dialog"
  id="background"
>
  <button
    class="absolute top-1/4"
  >
    <img
      class="h-8 -rotate-180 hover:rotate-180 transition-all ease-out cursor-pointer"
      src="/focus-todo-list/src/assets/close-circle.svg"
    />
  </button>
  <div
    class="w-3/4 flex flex-col gap-4 border-2 p-8 rounded-lg bg-slate-400"
  >
    <button
      class="self-end"
    >
      <img
        class="h-8 hover:brightness-200"
        src="/focus-todo-list/src/assets/add-square.svg"
      />
    </button>
    <div
      class="flex justify-center items-center h-24  rounded-2xl text-white font-bold bg-slate-400"
    >
      All Done!
    </div>
  </div>
</div>
`;
